torchgeo/datasets/idtrees.py

```diff
@@ -0,0 +1,540 @@
+
+    def plot(
+        self,
+        sample: Dict[str, Tensor],
+        show_titles: bool = True,
+        suptitle: Optional[str] = None,
+        hsi_indices: Tuple[int, int, int] = (0, 1, 2),
+    ) -> plt.Figure:
+        """Plot a sample from the dataset.
+
+        Args:
+            sample: a sample returned by :meth:`__getitem__`
+            show_titles: flag indicating whether to show titles above each panel
+            suptitle: optional string to use as a suptitle
+            hsi_indices: tuple of indices to create HSI false color image
+
+        Returns:
+            a matplotlib Figure with the rendered sample
+        """
+        assert len(hsi_indices) == 3
+
+        def normalize(x: Tensor) -> Tensor:
+            return (x - x.min()) / (x.max() - x.min())
+
+        ncols = 3
+
+        hsi = normalize(sample["hsi"][hsi_indices, :, :]).permute((1, 2, 0)).numpy()
+        chm = normalize(sample["chm"]).permute((1, 2, 0)).numpy()
+
+        if "boxes" in sample:
+            labels = (
+                [self.idx2class[int(i)] for i in sample["label"]]
+                if "label" in sample
+                else None
+            )
+            image = draw_bounding_boxes(
+                image=sample["image"], boxes=sample["boxes"], labels=labels
+            )
+            image = image.permute((1, 2, 0)).numpy()
+        else:
+            image = sample["image"].permute((1, 2, 0)).numpy()
+
+        if "prediction_boxes" in sample:
+            ncols += 1
+            labels = (
+                [self.idx2class[int(i)] for i in sample["prediction_label"]]
+                if "prediction_label" in sample
+                else None
+            )
+            preds = draw_bounding_boxes(
+                image=sample["image"], boxes=sample["prediction_boxes"], labels=labels
+            )
+            preds = preds.permute((1, 2, 0)).numpy()
+
+        fig, axs = plt.subplots(ncols=ncols, figsize=(ncols * 10, 10))
+        axs[0].imshow(image)
+        axs[0].axis("off")
+        axs[1].imshow(hsi)
+        axs[1].axis("off")
+        axs[2].imshow(chm)
+        axs[2].axis("off")
+        if ncols > 3:
+            axs[3].imshow(preds)
+            axs[3].axis("off")
+
+        if show_titles:
+            axs[0].set_title("Ground Truth")
+            axs[1].set_title("Hyperspectral False Color Image")
+            axs[2].set_title("Canopy Height Model")
+            if ncols > 3:
+                axs[3].set_title("Predictions")
+
+        if suptitle is not None:
+            plt.suptitle(suptitle)
+
+        return fig
+
+    def plot_las(self, index: int, colormap: Optional[str] = None) -> None:
+        """Plot a sample point cloud at the index.
+
+        Args:
+            index: index to plot
+            colormap: a valid matplotlib colormap
+
+        Raises:
+            ImportError: if open3d is not installed
+        """
+        try:
+            import open3d  # noqa: F401
```

Comment:
Need to add optional dependency on this. Someone will also have to add a Spack package.

Reply:
I think open3d stable is broken on pypi right now. I had to install directly from one of their dev wheels [here](http://www.open3d.org/docs/latest/getting_started.html). Might have to hack something in the github actions to get the tests to work.

Reply:
Open3d is on anaconda but it's under the `open3d-admin` channel, not `conda-forge`.

Reply:
Ok so looks like open3d is installable everywhere except on Python 3.9 via pypi. I'm going to add some logic to the tests action to account for this unless you have another suggestion.